### ISSUE TITLE
How many timesteps are needed at this moment to get the result?

### ISSUE THREAD

**Issue by Mimocro** [lucidrains/imagen-pytorch#31]
I trained the model with a loss of about 0.06 on a train, and here are some samples with the prompt "Milky Way in the shape of a dog":
100 timesteps no ema
![100 n](https://user-images.githubusercontent.com/67226986/172045259-50fdae07-3188-40bd-b17d-b439ff9b8442.png)
1000 timesteps no ema
![1000n](https://user-images.githubusercontent.com/67226986/172045253-cea9223b-a03c-47d5-8ee7-e0a8ba620a04.png)
10000 timesteps no ema
![10000 n](https://user-images.githubusercontent.com/67226986/172045257-f652232f-9ca5-495e-97ea-9790f892cbc6.png)
100 timesteps with ema
![100 ema](https://user-images.githubusercontent.com/67226986/172045258-4146e101-1963-44bb-8e96-961cc8703c5c.png)
1000 timesteps with ema
![1000 ema](https://user-images.githubusercontent.com/67226986/172045260-f930d4bf-de61-4c16-84fb-ad08b3e72480.png)
1000 timesteps with ema
![10000 ema](https://user-images.githubusercontent.com/67226986/172045256-562b0f83-02f4-4284-8c77-f5a49cba14a4.png)

I know the losses are too high for good results, but i expected at least something, and 10k steps looks a bit more complete. And the time spent on this is too long, 5 minutes for 1k steps and 50 minutes for 10k steps on kaggle's T4.
Also i train this model without ema, disabling or enabling this when i initializing trainer to load checkpoint for inference like 

`trainer = ImagenTrainer(imagen, use_ema = False)`
` trainer.load('./checkpoint/checkpoint.pt')`

My gpu quota is ended today and i can't try to continue training with ema to see how it's affect sampling with/without ema on it. Is it really will improve training speed (i need to wait about 15 epochs before losses decay by 0.001) and result not only noise samples with ema on?
Link to this checkpoint https://drive.google.com/file/d/1pq2OVRJuA2szQc9WMd8M7mkOtbtUCkv4/view?usp=sharing

**Comment by Mimocro** [comment]
Oh, sorry, i forget to train the second unet...
